neuralop/training/callbacks.py

```diff
@@ -633,7 +493,30 @@ def grad_explained(self):
             # update the modes and frequency dimensions
             self.grad_iter = 1
             self.accumulated_grad = torch.zeros_like(
-                self.state_dict['model'].fno_blocks.convs.weight[0])
+                model.fno_blocks.convs.weight[0])
             main_modes = incremental_final[0]
             modes_list = tuple([main_modes] * self.ndim)
-            self.state_dict['model'].fno_blocks.convs.n_modes = tuple(modes_list)
+            model.fno_blocks.convs.n_modes = tuple(modes_list)
+
+class LossBalancingCallback(Callback):
+    # source: https://arxiv.org/pdf/2308.08468
+    # Wang et al., "An Expert's Guide to Training Physics-Informed Neural Networks"
+    def __init__(self, alpha: float=0.9):
+        super().__init__()
+        self.alpha = alpha
+    
+    def compute_training_loss(self, out, sample, loss, model, **kwargs):
+        loss_vals = loss(out, **sample) # sumlossoutput
+        grad_norms = {}
+        total_grad_norm = 0.
+        for name, val in loss_vals.losses.items():
```

Comment:
We do *not* want to auto-balance every step since this is too expensive. We should be able to periodically update the weights of the `WeightedSumLoss` --- is this possible with this callback?